connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -577,15 +596,14 @@ private void resetDelay() {
         numToRevoke = floorTasks;
         for (WorkerLoad existing : existingWorkers) {
             Iterator<ConnectorTaskId> tasks = existing.tasks().iterator();
+            numToRevoke = existing.tasksSize() - ceilTasks;
```

Comment:
Isn't it possible that `numToRevoke` might be negative?

Reply:
yes it can be and it is captured under the next for loop condition.

Reply:
Can you explain a bit what you aim to achieve with this change here?

Reply:
Also, we apply this logic in tasks only. But why not in the connectors too, if it helps?

Reply:
@kkonstantine  we are trying to revoke the additional tasks assigned [if any] to a worker so that in the next stage the revoked tasks can be re assigned to the new workers that get added to the group.  Yes we need to apply this to connectors as well, I will update the PR for the changes .